crates/noirc_driver/src/contract.rs

```diff
@@ -33,12 +21,38 @@ pub enum ContractFunctionType {
 }
 
 #[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct CompiledContract {
     /// The name of the contract.
     pub name: String,
     /// Each of the contract's functions are compiled into a separate `CompiledProgram`
-    /// stored in this `BTreeMap`.
-    pub functions: BTreeMap<String, ContractFunction>,
+    /// stored in this `Vector`.
+    pub functions: Vec<ContractFunction>,
+}
+
+/// Each function in the contract will be compiled
+/// as a separate noir program.
+///
+/// A contract function unlike a regular Noir program
+/// however can have addition properties.
+/// One of these being a function type.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ContractFunction {
+    pub name: String,
+
+    pub function_type: ContractFunctionType,
+
+    #[serde(flatten)]
+    pub abi: Abi,
```

Comment:
`returnType` here is an array for some reason? 

https://github.com/AztecProtocol/aztec3-packages/blob/fec3dd74e3a1d5c087992d5d9f878ca1b37e969d/yarn-project/noir-contracts/src/examples/zk_token_contract.json#L37